File: java/src/test/java/kata/FindAndOrganizeNewPhotosTest.java
package kata;


import org.apache.commons.io.FileUtils;
import org.apache.commons.io.filefilter.TrueFileFilter;
import org.junit.Assert;
import org.junit.Test;
import org.junit.internal.runners.statements.Fail;

import java.io.File;
import java.io.IOException;

public class FindAndOrganizeNewPhotosTest {

    public static final String INPUT_DIRECTORY = "../validate/testData/input/";
    public static final String TARGET_DIRECTORY = "../validate/target/";
    public static final String CURRENT_PHOTOS_DIRECTORY = "../validate/testData/currentPhotos/";

    @Test
    public void testYearAndMonthAreDerivedFromFileName() {
        PhotoSorter photoSorter = new PhotoSorter();
        Assert.assertEquals("201701", photoSorter.extractYearAndMonthFromFileName("resources/testData/input/20170101_123456.jpg"));
        Assert.assertEquals("201702", photoSorter.extractYearAndMonthFromFileName("20170201_123456.jpg"));
        Assert.assertEquals("201703", photoSorter.extractYearAndMonthFromFileName("resources/testData/input/IMG-20170301-WA0000.jpg"));
    }

    @Test
    public void testYearAndMonthAreEmptyIfFileNameDoesntContainADate() {
        PhotoSorter photoSorter = new PhotoSorter();
        Assert.assertEquals("", photoSorter.extractYearAndMonthFromFileName(""));
        Assert.assertEquals("", photoSorter.extractYearAndMonthFromFileName("aap"));
    }

    @Test
    public void testFiveFilesAreReadFromTestInput() {
        Assert.assertEquals(5, new PhotoSorter().findFiles(INPUT_DIRECTORY, "").size());
    }

    @Test
    public void testOutputContainsCorrectFiles() {
        PhotoSorter photoSorter = new PhotoSorter();
        try {
            photoSorter.copyFilesToDirectories(INPUT_DIRECTORY, TARGET_DIRECTORY);
        } catch (IOException e) {
            Assert.fail("Failed to create " + TARGET_DIRECTORY);
        }
        Assert.assertTrue(new File(TARGET_DIRECTORY + "201701/20170101_123456.jpg").exists());
        Assert.assertTrue(new File(TARGET_DIRECTORY + "201701/20170102_123456.jpg").exists());
        Assert.assertTrue(new File(TARGET_DIRECTORY + "201701/IMG-20170101-WA0000.jpg").exists());
        Assert.assertTrue(new File(TARGET_DIRECTORY + "201702/20170201_123456.jpg").exists());
        Assert.assertTrue(new File(TARGET_DIRECTORY + "201702/IMG-20170201-WA0000.jpg").exists());
    }

    @Test
    public void testPhotosInCurrentFolderAreNotCopied() {
        PhotoSorter photoSorter = new PhotoSorter();
        try {
            photoSorter.copyFilesToDirectories(INPUT_DIRECTORY, TARGET_DIRECTORY, CURRENT_PHOTOS_DIRECTORY);
        } catch (IOException e) {
            Assert.fail("Failed to create " + TARGET_DIRECTORY);
        }
        Assert.assertEquals(2, new File(TARGET_DIRECTORY + "/201701").listFiles().length);
    }
}

File: x/test/find_and_organize_new_photos_test.py
from kata import FindAndOrganizeNewPhotos
import unittest
import os


CURRENTPHOTOSDIRECTORY = "../../validate/testData/currentPhotos"

SOURCEDIRECTORY = "../../validate/testData/input"

OUTPUTDIRECTORY = "../target"


class FindAndOrganizeNewPhotosTest(unittest.TestCase):
    def testGetJpgFilesReturnsFiveFiles(self):
        jpg_files = FindAndOrganizeNewPhotos().get_jpg_files(SOURCEDIRECTORY)
        self.assertEqual(5, len(jpg_files))

    def testExtractDateFromFileName(self):
        self.assertEqual('201701', FindAndOrganizeNewPhotos().extract_date_from_file_name("20170101_123456.jpg"))
        self.assertEqual('201701', FindAndOrganizeNewPhotos().extract_date_from_file_name("IMG-20170101_WA0000.jpg"))
        self.assertEqual('', FindAndOrganizeNewPhotos().extract_date_from_file_name("IMG-xx170101_WA0000.jpg"))
        self.assertEqual('', FindAndOrganizeNewPhotos().extract_date_from_file_name("aap-20170101_WA0000.jpg"))
        self.assertEqual('', FindAndOrganizeNewPhotos().extract_date_from_file_name("IM-20170101_WA0000.jpg"))
        self.assertEqual('', FindAndOrganizeNewPhotos().extract_date_from_file_name("aap"))
        self.assertEqual('201702', FindAndOrganizeNewPhotos().extract_date_from_file_name('../../validate/testData/input/IMG-20170201-WA0000.jpg'))

    def testFilesAreCopiedToTargetDir(self):
        FindAndOrganizeNewPhotos().create_empty_target_folder(OUTPUTDIRECTORY)
        FindAndOrganizeNewPhotos().copy_new_photos_to_target_folder(SOURCEDIRECTORY, OUTPUTDIRECTORY)
        target_files = FindAndOrganizeNewPhotos().get_jpg_files(OUTPUTDIRECTORY)
        self.assertEqual(5, len(target_files))
        self.assertTrue(os.path.exists(OUTPUTDIRECTORY + "/201701/20170101_123456.jpg"))

    def testFilesInCurrentDirAreIgnoredIfCurrentDirIsSet(self):
        FindAndOrganizeNewPhotos().create_empty_target_folder(OUTPUTDIRECTORY)
        FindAndOrganizeNewPhotos().copy_new_photos_to_target_folder(SOURCEDIRECTORY, OUTPUTDIRECTORY, current_photos_directory=CURRENTPHOTOSDIRECTORY)
        target_files = FindAndOrganizeNewPhotos().get_jpg_files(OUTPUTDIRECTORY)
        self.assertEqual(4, len(target_files))
        self.assertFalse(os.path.exists(OUTPUTDIRECTORY + "/201701/20170102_123456.jpg"))


if __name__ == '__main__':
    unittest.main()
